server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationIT.java

```diff
@@ -122,5 +172,61 @@ public void testReplicaSetupAfterPrimaryIndexesDocs() throws Exception {
         ensureGreen(INDEX_NAME);
         assertHitCount(client(nodeA).prepareSearch(INDEX_NAME).setSize(0).setPreference("_only_local").get(), 2);
         assertHitCount(client(nodeB).prepareSearch(INDEX_NAME).setSize(0).setPreference("_only_local").get(), 2);
+        assertSegmentStats(REPLICA_COUNT);
+    }
+
+    private void assertSegmentStats(int numberOfReplicas) {
+        client().admin().indices().segments(new IndicesSegmentsRequest(), new ActionListener<>() {
+            @Override
+            public void onResponse(IndicesSegmentResponse indicesSegmentResponse) {
+                final Map<String, IndexSegments> indices = indicesSegmentResponse.getIndices();
+                for (Map.Entry<String, IndexSegments> indexSegmentsEntry : indices.entrySet()) {
+                    final IndexSegments value = indexSegmentsEntry.getValue();
+                    final Map<Integer, IndexShardSegments> shardGroup = value.getShards();
+                    for (IndexShardSegments shardEntry : shardGroup.values()) {
+                        // get a list of segments across the whole replication group.
+                        final ShardSegments[] shardSegments = shardEntry.getShards();
```

Comment:
Nit - You can compress this using streams.

Option 1 - get a list of arrays to iterate over:

```
List<ShardSegments[]> shardSegmentsList = 
indicesSegmentResponse.getIndices().values().stream() // get list of IndexSegments
.flatMap(is -> is.getShards().values().stream()) // Map to shard group
.map(IndexShardSegments::getShards) // get list of segments across group
.collect(Collectors.toList());
```

Option 2 (less readable) - Generate a list of maps directly

```
List<Map<Boolean, List<ShardSegments>>> shardGroupList = 
indicesSegmentResponse.getIndices().values().stream() // get list of IndexSegments
.flatMap(is -> is.getShards().values().stream()) // Map to shard group
.map(IndexShardSegments::getShards) // get subarray of segments across group

.map(s -> Arrays.stream(s)) // convert each subarray to stream
.map(s -> s.collect(Collectors.groupingBy(
    ss -> ss.getShardRouting().primary() // group each sublist by primary
)))
.collect(Collectors.toList());
```

Reply:
Updated with option 1.  Thanks for this.